e2e/flags.go

```diff
@@ -95,6 +91,10 @@ var KCC = flag.Bool("kcc", false,
 var GceNode = flag.Bool("gcenode", false,
 	"If true, run test with 'gcenode' auth type.")
 
+// WorkloadIdentity enables running the e2e tests for 'gcpserviceaccount' and 'k8sserviceaccount' auth types
+var WorkloadIdentity = flag.Bool("workloadidentity", false,
```

Comment:
Does this imply we're going to add another CI job permutation for `--workload-identity`? It seems like it can be implied to run the WI tests when `TestCluster==gke` and `GceNode==false`

Reply:
These WI tests need to run when `--test-cluster=gke` and `--gcenode=false`, and `--git-provider=csr`. 
It would need a new CI job with these flags, similarly as how the gcenode CI is configured. We probably don't need a permutation to run on all channels.

Reply:
Ack we need to set `--git-provider=csr` specifically to run the git/WI tests, however there is still overlap for the `workloadidentity` auth types for other source providers (e.g. GAR with oci-sync/helm-sync).

Here I'm just talking about this new `workloadidentity` flag which means we need to create new testgrid permutations to specifically run any of the workload identity tests. I'm thinking it would be simpler to omit this flag and run the `workloadidentity` tests based on the existing options rather than adding new permuations.

Reply:
This is essentially the inverse of the `gcenode` flag, so having both seems like it might create redundant permutation

Reply:
I see. So you want oci/helm + wi to run with the current CI jobs, but only have a new job to run git + wi.

I was thinking all `workloadidentity` tests running on a separate job, including git, oci, and helm. We'll need a separate job anyway. This option runs all `workloadidentity` tests consistently.

This new flag allows us to select all `workloadidentity` tests for the new job, and skip them for other CI jobs.

Reply:
> This is essentially the inverse of the `gcenode` flag, so having both seems like it might create redundant permutation

It is not the inverse of the `gcenode` flag. We'll need 3 types of CI jobs:
- a job with the `--gcenode` flag that only runs the `ntopts.GCENodeTest` tests.
- a job with `--gcenode=false`, `--git-provider=local`, `--oci-provider=gar`  and `--helm-provider=gar` that runs  other testcases with local test-git-server. (oci/helm + wi tests won't be selected)
- a job with `--gcenode=false`, `--git-provider=csr`, `--oci-provider=gar`  and `--helm-provider=gar` that only runs the `ntopts.WorkloadIdentityTest` tests.

Reply:
To run all of the `workloadidentity` tests with this new permutation you would need to provide `--helm-provider=gar --oci-provider=gar --git-provider=csr --workloadidentity --test-cluster=gke`, which is already a lot of parameters. Also, it's already assumed for all of the other tests that if the helm/oci provider is GAR that it will default to workloadidentity auth.

Without this parameter, the only new job permutation we need is one that sets `--git-provider=csr`.

Reply:
Without this parameter, how do you filter out the git+wi test from other jobs?

Reply:
Based on whether the git provider supports workload identity, i.e. `RequireCloudSourceRepository`. This would be similar to how it's currently handled for helm/oci with `RequireHelmArtifactRegistry`/`RequireOCIArtifactRegistry`

Reply:
Right. The PR has it in place: https://github.com/GoogleContainerTools/kpt-config-sync/pull/1160/files#diff-8d37aad37a787f98c6132f35743b0be4045a4470a4f6554baa6015b7d6a0b66dR209.

Based on the existing Prow configuration (--git-provider=local), the git+wi test is selected and would fail after this PR because of `ntopts.RequireCloudSourceRepository(t)`.
We would need a way to filter out the git+wi test on existing CI jobs.

Reply:
If `--git-provider=local` I expect that these tests would be skipped rather than fail, right?

My understanding for the `ntopts.RequireCloudSourceRepository(t)` is that they run if `--git-provider=csr` and are skipped otherwise.

Reply:
Ah, I see. Somehow I thought these requireXXX flags would fail the test if it didn't match.
Then I agree, we don't need the flag. Will update the PR.

Reply:
updated